File: AEDS/aeds2/tp1/alteracaoaleatoria/AlteracaoAleatoria.java
import java.util.Scanner;
import java.lang.Math;
import java.util.Random;

public class AlteracaoAleatoria {

    private static Random gerador = new Random(4);
    
    //funcao para comparar strings e determinar se sao identicas
    public static boolean isEquals(String palavra1, String palavra2){
        int n = palavra1.length();
        boolean resp = true;
        // comparar tamanho das palavras 2 e 2
        if(n != palavra2.length()){
            resp = false;
        } else {
            for(int i = 0; i < n; i++){
                if(palavra1.charAt(i) != palavra2.charAt(i)){
                    resp = false;
                    i = n;
                 }
            }
        }
        return (resp);
    }

    // funcao para substituir caractere na String
    public static void substituirString(String frase){
        char letraAntiga = (char)('a' + (Math.abs(gerador.nextInt())%26));
        char letraNova = (char)('a' + (Math.abs(gerador.nextInt())%26));
        String novaFrase = "";
        
        for(int i = 0; i < frase.length(); i++){
            if(frase.charAt(i) == letraAntiga){
                novaFrase += letraNova;
            } else{
                 novaFrase += frase.charAt(i);
            }
        }
        System.out.println(novaFrase);
    }

    //metodo principal
    public static void main (String[] args){
        Scanner sc = new Scanner(System.in, "UTF-8");
        String frase = sc.nextLine();
    
        while(!isEquals(frase,"FIM")){
            substituirString(frase);
            frase = sc.nextLine();
        }
        sc.close();
    }
}

File: 2° semestre/aeds2/tp1/alteracaoaleatoria/c/AlteracaoAleatoria.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>  // pra uso do rand()

// funcao para determinar tamanho da palavra
int str_len(char *palavra){
    int length = 0;
	while(*(palavra+length)){ 
		length++;
	}
	return(length);
}

//funcao para comparar strings 
bool str_cmp(char palavra1[], char palavra2[]){
    bool resp = true;
    int length = str_len(palavra1);

    if(length != str_len(palavra2)){
        resp = false;    
    } else {
        for(int i = 0; i < length; i++){
            if(palavra1[i] != palavra2[i]){
               resp = false;
               i = length;
            } 
        }
    }
    return (resp);
}
    
// funcao para substituir caractere na String
void substituirString(char s[], int i,char novaFrase[], char letraAntiga, char letraNova){
    //char novaFrase[500] = "";
   
	if(i == str_len(s)){
		novaFrase[i] ='\0';
		printf("%s\n", novaFrase);
	} else {
    	if (s[i] == letraAntiga) {
        	novaFrase[i] = letraNova;
    	} else {
        	novaFrase[i] = s[i];
		}
		substituirString(s, i + 1, novaFrase, letraAntiga, letraNova);
    }
}

//metodo principal
int main (){
    char s[500];
	char novaFrase[500];
	srand(4); //seed 4

    scanf(" %[^\n]", s);  
    
    while(!str_cmp(s, "FIM")){
		char letraAntiga = 'a' + (rand() % 26);
    	char letraNova = 'a' + (rand() % 26);  

        substituirString(s, 0, novaFrase, letraAntiga, letraNova);
        scanf(" %[^\n]", s);  
	}
    return 0;
}
